### ISSUE TITLE
Enhance data input to clear certain parts of the timestamp.

### ISSUE THREAD

**Issue by discordier** [MetaModels/attribute_timestamp#27]
## Issue description

When editing timestamps with formats where parts are missing (Year, day, month, hour, minute, second), PHP does initialize it with the current `time()` value.
This is correct and good from PHP side, as formats like `+1 week` are perfectly legal and useful, yet it will break our filtering.

So we need some way to restrict time or date to "zero" in order to allow valid filtering and omit hickups like discussed in MetaModels/filter_fromto#9.

## Proposed solution
The idea I had is to add some input screen configuration setting where the user can select one of three possible settings:
1. do nothing (default and current behaviour)
2. clear data part (will restrict the date to 01.01.1970 internally).
3. clear time part (will restrict the time to 00:00:00 internally).
 
I'd put this setting into the input screen definition of the attribute instance to allow variation of this in the concrete input screens.

Implementation would then be to [change the current implementation](https://github.com/MetaModels/attribute_timestamp/blob/7b22eede3065c7cdb7a1e7529668e133d596bb33/src/MetaModels/Attribute/Timestamp/BackendSubscriber.php#L68) something along the lines of:
```php
    public function handleEncodePropertyValueFromWidget(EncodePropertyValueFromWidgetEvent $event)
    {
        $attribute = $this->getSupportedAttribute($event);
        if (!$attribute) {
            return;
        }

        $date       = \DateTime::createFromFormat($attribute->getDateTimeFormatString(), $event->getValue());
        $properties = $event->getEnvironment()->getDataDefinition()->getPropertiesDefinition();
        $property   = $properties->getProperty($event->getProperty());
        $extra      = $property->getExtra();
        if (isset($extra['clear'])) {
            switch ($extra['clear']) {
                case 'time':
                    $date->setTime(0, 0, 0);
                    break;
                case 'date':
                    $date->setDate(0, 0, 0);
                    break;
                default:
            }
        }

        if ($date) {
            $event->setValue($date->getTimestamp());
        }
    }
```

Of course we would also have to add such setting to the fromto filter regarding dates, as otherwise date filtering would begin and end "now" and therefore might not find items due to the offset.

**Comment by zonky2** [comment]
sounds good for me...

**Comment by discordier** [comment]
Addendum: We also should move the `timeformat` setting from the attribute to the input screen setting to allow different input formats per input screen (different formats for different user groups from different countries i.e.).

**Comment by zonky2** [comment]
Isn't it a simpler way if we make our own comparison for date, date-time, time as our own `getIdsFiltered`. Here you can convert and compare the timestamps into the corresponding formats.

**Comment by bezin** [comment]
Hi there, 

what is the status on this one? For our current project it would be super handy to have times stored without the date part (i.e. 11:00am not as "Date of Modification 11:00am" but as "01.01.1970 11:00am").

**Comment by zonky2** [comment]
implemented in MM 2.2
